packages/editor-ui/src/stores/assistant.store.ts

```diff
@@ -498,7 +504,7 @@ export const useAssistantStore = defineStore(STORES.ASSISTANT, () => {
 			},
 			(msg) => onEachStreamingMessage(msg, id),
 			() => onDoneStreaming(id),
-			(e) => handleServiceError(e, id),
+			(e) => handleServiceError(e, id, async () => await initErrorHelper(context)),
```

Comment:
Can you explain why you could not simplify it to `() => initErrorHelper(context)`? I think you said it's related to the tests. 
Happy to pair on this to figure it out together.

Reply:
Ah so the issue was that I got this eslint error https://typescript-eslint.io/rules/promise-function-async/ and mistook it for a TS compilation error in haste. Our use case seems very reasonable despite this, though I don't think it hurts to be explicit about the callback not executing synchronously, which is probably another argument for the eslint rule in the first place.

Indeed for the test I could still await on the returned promise from the synchronous function, so disregard that.